pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -968,21 +969,67 @@ func (a *ApplicationBackupController) backupResources(
 	// Always backup optional resources. When restorting they need to be
 	// explicitly added to the spec
 	objectMap := stork_api.CreateObjectsMap(backup.Spec.IncludeResources)
-	allObjects, err := a.resourceCollector.GetResources(
-		backup.Spec.Namespaces,
-		backup.Spec.Selectors,
-		objectMap,
-		optionalBackupResources,
-		true)
-	if err != nil {
-		log.ApplicationBackupLog(backup).Errorf("Error getting resources: %v", err)
-		return err
+	namespacelist := backup.Spec.Namespaces
+	// GetResources takes more time, if we have more number of namespaces
+	// So, submitting it in batches and in between each batch,
+	// updating the LastUpdateTimestamp to show that backup is progressing
+	var allObjects []runtime.Unstructured
+	allObjects = make([]runtime.Unstructured, 0)
+	for i := 0; i < len(namespacelist); i += backupResourcesBatchCount {
+		batch := namespacelist[i:min(i+backupResourcesBatchCount, len(namespacelist))]
+		objects, err := a.resourceCollector.GetResources(
+			batch,
+			backup.Spec.Selectors,
+			objectMap,
+			optionalBackupResources,
+			true)
+		if err != nil {
+			log.ApplicationBackupLog(backup).Errorf("Error getting resources: %v", err)
+			return err
+		}
+		allObjects = append(allObjects, objects...)
+		// Do a dummy update to the backup CR to update only the last update timestamp
+		namespacedName := types.NamespacedName{}
+		namespacedName.Namespace = backup.Namespace
+		namespacedName.Name = backup.Name
+		for i := 0; i < maxRetry; i++ {
+			err := a.client.Get(context.TODO(), namespacedName, backup)
+			if err != nil {
+				time.Sleep(retrySleep)
+				continue
+			}
+			backup.Status.LastUpdateTimestamp = metav1.Now()
+			err = a.client.Update(context.TODO(), backup)
+			if err != nil {
+				time.Sleep(retrySleep)
+				continue
+			} else {
+				break
+			}
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
This will never be hit

Reply:
Fixed. using error variable which was declared outside the for loop, so that check outside the for loop will use correct err variable.